File: source/1436.旅行终点站.cs
/*
 * @lc app=leetcode.cn id=1436 lang=csharp
 *
 * [1436] 旅行终点站
 */
using csharpnaming;
// @lc code=start
public class Solution {
    public string DestCity(IList<IList<string>> paths) {
        var startCities = new HashSet<string>();
        var endCities = new HashSet<string>();

        foreach(var path in paths) {
            startCities.Add(path[0]);
            endCities.Add(path[1]);
        }

        foreach(var city in endCities) {
            if (startCities.Contains(city) == false) {
                return city;
            }
        }

        return string.Empty;
    }
}
// @lc code=end

File: source/1436.旅行终点站.cpp
/*
 * @lc app=leetcode.cn id=1436 lang=cpp
 *
 * [1436] 旅行终点站
 */
#include "cpp_includes.h"
// @lc code=start
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> startCities;
        unordered_set<string> endCities;

        for (int i = 0; i < paths.size(); i++) {
            startCities.insert(paths[i][0]);
            endCities.insert(paths[i][1]);
        }

        for (auto dest : endCities) {
            if (!startCities.count(dest)){
                return dest;
            }
        }

        return "";
    }
};
// @lc code=end

